mongoreplay/mongo_op_handler.go

```diff
@@ -312,20 +312,37 @@ func (bidi *bidi) handleStreamStateOutOfSync(stream *stream) {
 // stream's and tries to create whole protocol messages from them.
 func (bidi *bidi) streamOps() {
 	bidi.logvf(Info, "starting")
-	for {
+	ch0 := bidi.streams[0].reassembled
+	ch1 := bidi.streams[1].reassembled
+	for ch0 != nil || ch1 != nil {
 		var reassemblies []tcpassembly.Reassembly
 		var reassembliesStream int
 		var ok bool
 		select {
-		case reassemblies, ok = <-bidi.streams[0].reassembled:
+		case reassemblies, ok = <-ch0:
 			reassembliesStream = 0
-		case reassemblies, ok = <-bidi.streams[1].reassembled:
+			if !ok {
+				ch0 = nil
+			}
+		case reassemblies, ok = <-ch1:
 			reassembliesStream = 1
+			if !ok {
+				ch1 = nil
+			}
 		}
+		stream := bidi.streams[reassembliesStream]
 		if !ok {
-			break
+			if !stream.opTimeStamp.IsZero() {
```

Comment:
The reason I have explicit closes going both ways right now is to get around the fact that these "close" recordedops always use the connectionString (and not the reversed connection string) in play.go. Having closes for both directions means that it is guaranteed to close the connection.

However I'd be happy to make this just one "close" recordedop, if play.go used something like the connectionNumber to identify the session?